src/explosion.cpp

```diff
@@ -373,90 +374,87 @@ static std::vector<tripoint> shrapnel( const tripoint &src, int power,
                  ( visited_cache, obstacle_cache, src.xy(), 0, initial_cloud );
 
     // Now visited_caches are populated with density and velocity of fragments.
-    for( int x = start.x; x < end.x; x++ ) {
```

Comment:
The original loop is exclusive of end and the new loop is inclusive of end, this is causing memory access errors.

Reply:
Good point. I shall fix this.